clang/tools/c-index-test/JSONAggregation.cpp

```diff
@@ -108,6 +108,34 @@ class Aggregator {
     return StringRef(buf, str.size());
   }
 
+  StringRef escapeForJson(StringRef str) {
+    char *buf = Allocator.Allocate<char>(str.size() * 2);
+
+    size_t i = 0;
+    for (char c : str) {
+      switch (c) {
+      case '"':
+        buf[i++] = '\\';
+        buf[i++] = '"';
+        break;
+      case '\\':
+        buf[i++] = '\\';
+        buf[i++] = '\\';
+        break;
+      case '\n':
+        buf[i++] = '\\';
+        buf[i++] = 'n';
+        break;
+      default:
+        buf[i++] = c;
+      }
+    }
+
+    buf[i] = '\0';
+
+    return StringRef(buf, str.size());
+  }
```

Comment:
You might want to use `llvm/Support/JSON.h` and do something like
```
OS.indent(6) << "\"kind\": " << llvm::json::Value(symInfo.Kind) << ",\n";
```
so you don't have to re-implement JSON quoting here. Ideally the whole code should use it, but a change like this is a good start.